Difficulty Compiling for STM32

So a bit of a n00b, but I'm running into troubles creating a new project
<details>
<summary>error logs</summary>
```bash
libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol WWDG
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol RTC
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol FLASH
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol RCC
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol EXTI0_1
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol EXTI2_3
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol EXTI4_15
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol DMA1_CHANNEL1
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol DMA1_CHANNEL2_3
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol DMA1_CHANNEL4_5
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol ADC1
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM1_BRK_UP_TRG_COM
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM1_CC
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM3
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM6
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM7
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM14
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM15
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM16
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM17
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol I2C1
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol I2C2
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol SPI1
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol SPI2
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol USART1
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol USART2
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol USART3_4
          libstm32_metapac-bd62a65356f59473.rlib(stm32_metapac-bd62a65356f59473.stm32_metapac.160da54d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol USB
          libcortex_m_rt-e3b58d0a626900a1.rlib(cortex_m_rt-e3b58d0a626900a1.cortex_m_rt.2fb0f525-cgu.3.rcgu.o) : error LNK2001: unresolved external symbol Reset
          libcortex_m_rt-e3b58d0a626900a1.rlib(cortex_m_rt-e3b58d0a626900a1.cortex_m_rt.2fb0f525-cgu.3.rcgu.o) : error LNK2001: unresolved external symbol NonMaskableInt
          libcortex_m_rt-e3b58d0a626900a1.rlib(cortex_m_rt-e3b58d0a626900a1.cortex_m_rt.2fb0f525-cgu.3.rcgu.o) : error LNK2001: unresolved external symbol HardFaultTrampoline
          libcortex_m_rt-e3b58d0a626900a1.rlib(cortex_m_rt-e3b58d0a626900a1.cortex_m_rt.2fb0f525-cgu.3.rcgu.o) : error LNK2001: unresolved external symbol MemoryManagement
          libcortex_m_rt-e3b58d0a626900a1.rlib(cortex_m_rt-e3b58d0a626900a1.cortex_m_rt.2fb0f525-cgu.3.rcgu.o) : error LNK2001: unresolved external symbol BusFault
          libcortex_m_rt-e3b58d0a626900a1.rlib(cortex_m_rt-e3b58d0a626900a1.cortex_m_rt.2fb0f525-cgu.3.rcgu.o) : error LNK2001: unresolved external symbol UsageFault
          libcortex_m_rt-e3b58d0a626900a1.rlib(cortex_m_rt-e3b58d0a626900a1.cortex_m_rt.2fb0f525-cgu.3.rcgu.o) : error LNK2001: unresolved external symbol SVCall
          libcortex_m_rt-e3b58d0a626900a1.rlib(cortex_m_rt-e3b58d0a626900a1.cortex_m_rt.2fb0f525-cgu.3.rcgu.o) : error LNK2001: unresolved external symbol DebugMonitor
          libcortex_m_rt-e3b58d0a626900a1.rlib(cortex_m_rt-e3b58d0a626900a1.cortex_m_rt.2fb0f525-cgu.3.rcgu.o) : error LNK2001: unresolved external symbol PendSV
          libcortex_m_rt-e3b58d0a626900a1.rlib(cortex_m_rt-e3b58d0a626900a1.cortex_m_rt.2fb0f525-cgu.3.rcgu.o) : error LNK2001: unresolved external symbol SysTick
```
</details>

Here are my dependencies

```rust
[dependencies]
stm32-metapac = "7.0.0"
cortex-m = { version = "0.7.6", features = ["critical-section-single-core"] }
cortex-m-rt = "0.7.1"
cortex-m-semihosting = "0.3.3"
panic-halt = "0.2.0"
embassy-executor = {version = "0.2.0", git = "https://github.com/embassy-rs/embassy", features = ["defmt", "integrated-timers"] }
embassy-time = {version = "0.1.0", git = "https://github.com/embassy-rs/embassy", features = ["defmt", "defmt-timestamp-uptime"] }
embassy-stm32 = {version = "0.1.0", git = "https://github.com/embassy-rs/embassy.git", features = ["defmt","stm32f070cb"]}
```

My kneejerk reaction is that `embassy-stm32` and `cortex-m-rt` are missing additional dependencies, or some additional toml settings, which I've tried to remediate by pulling them into my dependencies as well. However, I have no idea what seems to be wrong with my current setup.

Quick update:

I decided to switch over to nightly, redownloaded the STM32 v6m compiler, and still got a very similar message, this time with the full context:

```bash
error: linking with `link.exe` failed: exit code: 1120
  |
  = note: "D:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" "C:\\Users\\Me\\AppData\\Local\\Temp\\rustcN4WX28\\symbols.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.0.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.1.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.10.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.11.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.12.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.13.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.14.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.15.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.2.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.3.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.4.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.5.rcgu.o" 
"C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.6.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.7.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.8.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.build_script_build.81c1c68622bd59a9-cgu.9.rcgu.o" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.1859h6dhhcwkefk9.rcgu.o" "/LIBPATH:C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps" "/LIBPATH:C:\\Users\\Me\\.cargo\\registry\\src\\index.crates.io-6f17d22bba15001f\\stm32-metapac-7.0.0/src/chips/stm32f070cb" "/LIBPATH:C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\cortex-m-rt-2ebccca765b4c1e1\\out" "/LIBPATH:C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps\\libstm32_metapac-9912149c59a1d024.rlib" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps\\libcortex_m-d2eb502326c64cda.rlib" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps\\libembedded_hal-1f47a0238fb4ac28.rlib" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps\\libvoid-6b3aa605112187c1.rlib" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps\\libnb-38e8662f8b5aab55.rlib" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps\\libnb-d90d895c5489634b.rlib" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps\\libvolatile_register-52a8f9c9d1491acf.rlib" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps\\libvcell-061b6a584b814dc5.rlib" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps\\libbare_metal-eafd81ca5e1206e6.rlib" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps\\libcortex_m_rt-4faad5b4cc4830e8.rlib" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps\\libquote-1c5a5b99f2b01435.rlib" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps\\libproc_macro2-224ef92cf7a8e297.rlib" "C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\deps\\libunicode_ident-fb44f0087c2a31dc.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libproc_macro-9e71230232747fa7.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd-f06e7c156c07babf.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libpanic_unwind-66d3788cb65295c2.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_demangle-15be6ae923696459.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd_detect-ccd246deb449e005.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libhashbrown-434e22ac2d88ab17.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_alloc-591ae27697943a49.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libminiz_oxide-9633500de6d1ff03.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libadler-a3b7bf1c07db541f.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libunwind-07e6fc7575a725e2.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcfg_if-5f5b866d8ed0bc9a.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liblibc-3983930a0c9e59c9.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liballoc-8b5016f268855328.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_core-69892fa511d5a698.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcore-798bc7c2d986ac97.rlib" "C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-fe5d904068e00acb.rlib" "kernel32.lib" "advapi32.lib" "bcrypt.lib" "kernel32.lib" "ntdll.lib" "userenv.lib" "ws2_32.lib" "kernel32.lib" "msvcrt.lib" "/NXCOMPAT" "/LIBPATH:C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "/OUT:C:\\Users\\Me\\Documents\\embed-rust\\app\\target\\debug\\build\\embassy-stm32-58942b6ee258a7a6\\build_script_build-58942b6ee258a7a6.exe" "/OPT:REF,NOICF" "/DEBUG" "/NATVIS:C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:C:\\Users\\Me\\.rustup\\toolchains\\nightly-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libstd.natvis"
  = note: libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol WWDG
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol RTC
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol FLASH
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol RCC
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol EXTI0_1
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol EXTI2_3
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol EXTI4_15
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol DMA1_CHANNEL1
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol DMA1_CHANNEL2_3
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol DMA1_CHANNEL4_5
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol ADC1
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM1_BRK_UP_TRG_COM
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM1_CC
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM3
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM6
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM7
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM14
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM15
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM16
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol TIM17
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol I2C1
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol I2C2
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol SPI1
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol SPI2
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol USART1
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol USART2
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol USART3_4
          libstm32_metapac-9912149c59a1d024.rlib(stm32_metapac-9912149c59a1d024.stm32_metapac.2823805c8cf4806d-cgu.4.rcgu.o) : error LNK2001: unresolved external symbol USB
          libcortex_m_rt-4faad5b4cc4830e8.rlib(cortex_m_rt-4faad5b4cc4830e8.cortex_m_rt.85cf949c9bb5a779-cgu.0.rcgu.o) : error LNK2001: unresolved external symbol Reset
          libcortex_m_rt-4faad5b4cc4830e8.rlib(cortex_m_rt-4faad5b4cc4830e8.cortex_m_rt.85cf949c9bb5a779-cgu.0.rcgu.o) : error LNK2001: unresolved external symbol NonMaskableInt
          libcortex_m_rt-4faad5b4cc4830e8.rlib(cortex_m_rt-4faad5b4cc4830e8.cortex_m_rt.85cf949c9bb5a779-cgu.0.rcgu.o) : error LNK2001: unresolved external symbol HardFaultTrampoline
          libcortex_m_rt-4faad5b4cc4830e8.rlib(cortex_m_rt-4faad5b4cc4830e8.cortex_m_rt.85cf949c9bb5a779-cgu.0.rcgu.o) : error LNK2001: unresolved external symbol MemoryManagement
          libcortex_m_rt-4faad5b4cc4830e8.rlib(cortex_m_rt-4faad5b4cc4830e8.cortex_m_rt.85cf949c9bb5a779-cgu.0.rcgu.o) : error LNK2001: unresolved external symbol BusFault
          libcortex_m_rt-4faad5b4cc4830e8.rlib(cortex_m_rt-4faad5b4cc4830e8.cortex_m_rt.85cf949c9bb5a779-cgu.0.rcgu.o) : error LNK2001: unresolved external symbol UsageFault
          libcortex_m_rt-4faad5b4cc4830e8.rlib(cortex_m_rt-4faad5b4cc4830e8.cortex_m_rt.85cf949c9bb5a779-cgu.0.rcgu.o) : error LNK2001: unresolved external symbol SVCall
          libcortex_m_rt-4faad5b4cc4830e8.rlib(cortex_m_rt-4faad5b4cc4830e8.cortex_m_rt.85cf949c9bb5a779-cgu.0.rcgu.o) : error LNK2001: unresolved external symbol DebugMonitor
          libcortex_m_rt-4faad5b4cc4830e8.rlib(cortex_m_rt-4faad5b4cc4830e8.cortex_m_rt.85cf949c9bb5a779-cgu.0.rcgu.o) : error LNK2001: unresolved external symbol PendSV
          libcortex_m_rt-4faad5b4cc4830e8.rlib(cortex_m_rt-4faad5b4cc4830e8.cortex_m_rt.85cf949c9bb5a779-cgu.0.rcgu.o) : error LNK2001: unresolved external symbol SysTick
          C:\Users\Me\Documents\embed-rust\app\target\debug\build\embassy-stm32-58942b6ee258a7a6\build_script_build-58942b6ee258a7a6.exe : fatal error LNK1120: 38 unresolved externals


error: could not compile `embassy-stm32` (build script) due to previous error
```

I updated

```rust
[package]
...
edition = "2021"
...
```

and that seemed to change _something_

```bash
error[E0433]: failed to resolve: unresolved import
 --> C:\Users\Me\Documents\embed-rust\app\target\thumbv6m-none-eabi\debug\build\embassy-stm32-fdf8868461f25813\out/_generated.rs:1:28806
  |
1 | ...: usart :: CtsPin , USART4 , PB7 , 4u8) ; pin_trait_impl ! (crate :: usb :: DmPin , USB , PA11 , 0u8) ; pin_trait_impl ! (crate :: usb :: DpPin , USB , PA12 , 0u8) ; dma_trait_impl ! (crate :... 
  |                                                                         ^^^
  |                                                                         |
  |                                                                         unresolved import
  |                                                                         help: a similar path exists: `stm32_metapac::usb`

error[E0433]: failed to resolve: unresolved import
 --> C:\Users\Me\Documents\embed-rust\app\target\thumbv6m-none-eabi\debug\build\embassy-stm32-fdf8868461f25813\out/_generated.rs:1:28868
  |
1 | ...ate :: usb :: DmPin , USB , PA11 , 0u8) ; pin_trait_impl ! (crate :: usb :: DpPin , USB , PA12 , 0u8) ; dma_trait_impl ! (crate :: i2c :: TxDma , I2C1 , { channel : DMA1_CH2 } , ()) ; dma_tra... 
  |                                                                         ^^^
  |                                                                         |
  |                                                                         unresolved import
  |                                                                         help: a similar path exists: `stm32_metapac::usb`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `embassy-stm32` (lib) due to 2 previous errors
```

I completely removed the `edition` section, and that seems to have done something

<details>
<summary>trace</summary>

```rust
error[E0425]: cannot find function `__basepri_r` in module `crate::asm::inline`
 --> C:\Users\Me\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cortex-m-0.7.7\src\register\basepri.rs:6:15
  |
6 |     call_asm!(__basepri_r() -> u8)
  |               ^^^^^^^^^^^ not found in `crate::asm::inline`

error[E0425]: cannot find function `__basepri_w` in module `crate::asm::inline`
  --> C:\Users\Me\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cortex-m-0.7.7\src\register\basepri.rs:22:19
   |
22 |         call_asm!(__basepri_w(basepri: u8));
   |                   ^^^^^^^^^^^ not found in `crate::asm::inline`

error[E0425]: cannot find function `__basepri_max` in module `crate::asm::inline`
  --> C:\Users\Me\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cortex-m-0.7.7\src\register\basepri_max.rs:19:19
   |
19 |         call_asm!(__basepri_max(basepri: u8));
   |                   ^^^^^^^^^^^^^ not found in `crate::asm::inline`

error[E0425]: cannot find function `__faultmask_r` in module `crate::asm::inline`
  --> C:\Users\Me\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cortex-m-0.7.7\src\register\faultmask.rs:29:28
   |
29 |     let r: u32 = call_asm!(__faultmask_r() -> u32);
   |                            ^^^^^^^^^^^^^ not found in `crate::asm::inline`

error: invalid register `r0`: unknown register
   --> C:\Users\Me\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cortex-m-0.7.7\src\..\asm\inline.rs:197:24
    |
197 |     asm!("bkpt #0xab", inout("r0") nr, in("r1") arg, options(nomem, nostack, preserves_flags));
    |                        ^^^^^^^^^^^^^^

error: invalid register `r1`: unknown register
   --> C:\Users\Me\.cargo\registry\src\index.crates.io-6f17d22bba15001f\cortex-m-0.7.7\src\..\asm\inline.rs:197:40
    |
197 |     asm!("bkpt #0xab", inout("r0") nr, in("r1") arg, options(nomem, nostack, preserves_flags));
    |     
```
</details>

--- a/ci.sh
+++ b/ci.sh
@@ -46,6 +46,20 @@ cargo batch  \
     --- build --release --manifest-path embassy-rp/Cargo.toml --target thumbv6m-none-eabi --features nightly,unstable-traits \
     --- build --release --manifest-path embassy-rp/Cargo.toml --target thumbv6m-none-eabi --features nightly \
     --- build --release --manifest-path embassy-rp/Cargo.toml --target thumbv6m-none-eabi --features nightly,intrinsics \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,defmt,exti,time-driver-any,unstable-traits \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,defmt,exti,time-driver-any \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,defmt,time-driver-any \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,defmt,time-driver-any,unstable-traits \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,defmt,exti \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,defmt,exti,unstable-traits \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,defmt \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,nightly,defmt,exti,time-driver-any,unstable-traits \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,nightly,defmt,exti,time-driver-any \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,nightly,defmt,time-driver-any \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,nightly,defmt,time-driver-any,unstable-traits \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,nightly,defmt,exti \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,nightly,defmt,exti,unstable-traits \
+    --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv8m.main-none-eabihf --features stm32l552ze,nightly,defmt \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features nightly,stm32f410tb,defmt,exti,time-driver-any,unstable-traits \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features nightly,stm32f411ce,defmt,exti,time-driver-any,unstable-traits \
     --- build --release --manifest-path embassy-stm32/Cargo.toml --target thumbv7em-none-eabi --features nightly,stm32f413vh,defmt,exti,time-driver-any,unstable-traits \

--- a/embassy-stm32/src/lib.rs
+++ b/embassy-stm32/src/lib.rs
@@ -58,7 +58,7 @@ pub mod spi;
 pub mod tl_mbox;
 #[cfg(usart)]
 pub mod usart;
-#[cfg(all(usb, feature = "time"))]
+#[cfg(usb)]
 pub mod usb;
 #[cfg(otg)]
 pub mod usb_otg;

--- a/embassy-stm32/src/usb/usb.rs
+++ b/embassy-stm32/src/usb/usb.rs
@@ -7,7 +7,6 @@ use core::task::Poll;
 
 use embassy_hal_common::into_ref;
 use embassy_sync::waitqueue::AtomicWaker;
-use embassy_time::{block_for, Duration};
 use embassy_usb_driver as driver;
 use embassy_usb_driver::{
     Direction, EndpointAddress, EndpointAllocError, EndpointError, EndpointInfo, EndpointType, Event, Unsupported,
@@ -200,7 +199,10 @@ impl<'d, T: Instance> Driver<'d, T> {
                 w.set_fres(true);
             });
 
-            block_for(Duration::from_millis(100));
+            #[cfg(time)]
+            embassy_time::block_for(embassy_time::Duration::from_millis(100));
+            #[cfg(not(time))]
+            cortex_m::asm::delay(crate::rcc::get_freqs().sys.0 / 10);
 
             #[cfg(not(usb_v4))]
             regs.btable().write(|w| w.set_btable(0));

